app/src/main/java/com/github/se/eventradar/model/repository/user/FirebaseUserRepository.kt

```diff
@@ -144,11 +151,24 @@ class FirebaseUserRepository(db: FirebaseFirestore = Firebase.firestore) : IUser
       uid: String,
       folderName: String
   ): Resource<Unit> {
-    return Resource.Success(Unit)
+    val storageRef = Firebase.storage.reference.child("$folderName/$uid")
+    return try {
+      storageRef.putFile(selectedImageUri).await()
```

Comment:
I don't believe `putFile` throws any exceptions (see [here](https://firebase.google.com/docs/reference/android/com/google/firebase/storage/StorageReference#putFile(android.net.Uri))). You will have to use the value returned from function to see if the storage was successful.